src/amount_ext.rs

```diff
@@ -21,10 +25,53 @@ impl FromCliStr for Amount {
     }
 }
 
+fn parse_suffix(string: &str) -> Option<Result<Amount, anyhow::Error>> {
+    let final_value: f32;
+    if void_denominations(string) {
+        match string.rfind(char::is_numeric) {
+            Some(i) => {
+                let user_value = &string[..=i].parse::<f32>().unwrap();
+                if check_k(string) {
```

Comment:
a note on style: I prefer to just keep logic inline rather than split it into functions. It's also preferable to just use  a match statement and check for "k" at the end here (see commit I added).